tests/MongoDB.Driver.Examples/CausalConsistencyExamples.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Examples
+{
+    public class CausalConsistencyExamples
+    {
+        [Fact]
+        public void Causal_Consistency_Example_2()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            GetClient().GetDatabase("myDatabase").DropCollection("myCollection");
+
+            // Start Tunable consistency controls example
+            var connectionString = "mongodb://localhost/?readPreference=secondaryPreferred";
+
+            var client = new MongoClient(connectionString);
+            var database = client.GetDatabase("myDatabase");
+            var collection = database.GetCollection<BsonDocument>("myCollection");
+
+            // Start client session, which is causally consistent by default
+            using (var session = client.StartSession())
+            {
+                // Run causally related operations within the session
+                collection.InsertOne(session, new BsonDocument("name", "Tom"));
+                collection.UpdateOne(session,
+                    Builders<BsonDocument>.Filter.Eq("name", "Tom"),
+                    Builders<BsonDocument>.Update.Set("name", "John"));
+
+                using (var cursor = collection.Find(session, FilterDefinition<BsonDocument>.Empty).ToEnumerable().GetEnumerator())
+                {
+                    while (cursor.MoveNext())
+                    {
+                        var cur = cursor.Current;
+                    }
+                }
+            }
+            // End Tunable consistency controls example
+
+            var result = collection.Find("{}").FirstOrDefault();
+            RemoveIds(new[] { result });
+            result.Should().Be("{ name: 'John' }");
+        }
+
+        private IMongoClient GetClient()
```

Comment:
Maybe the method should be called "DropCollection" and do a bit more work?

Reply:
I will change this method to PrepareClient()

Reply:
As it was discussed in another PR, I've created two methods: CreateClient, DropCollection